activerecord/lib/active_record/associations/collection_proxy.rb

```diff
@@ -317,6 +317,13 @@ def build(attributes = {}, &block)
       end
       alias_method :new, :build
 
+      def insert_all(attributes, returning: nil, unique_by: nil)
+        relation_attributes = @association.insert_all_attributes
```

Comment:
Use `creation_attributes` to support polymorphic association.

https://github.com/rails/rails/blob/26b5e6284703432fc9022b215a221037fc05920d/activerecord/lib/active_record/associations/association.rb#L239-L251

Reply:
What do you think of using `scope_for_create` instead of `@association.creation_attributes`?

This would, of course, do _more_ than just include association's keys but would incorporate implicit constraints from `.where` clauses and `.create_with` values as well, wouldn't it?

Reply:
That's actually used at my approach at https://github.com/rails/rails/pull/35638.